server/api/reminders_api.js

```diff
@@ -0,0 +1,169 @@
+    console.log('New York :' + newYorkTime.format() + " Los Angeles :" + losAngeles.format() + " Central timeZone" + centralAmerica.format());
+
+    var deliverAt = newYorkTime.format() ; // defaulting to eastern timezone
+
+    if (fields.timeZone == "Eastern") {deliverAt = newYorkTime.format()} ;
+    if (fields.timeZone == "Central") {deliverAt = centralAmerica.format()} ;
+    if (fields.timeZone == "Pacific") {deliverAt = losAngeles.format()} ;
+    if (fields.timeZone == "IST") {deliverAt = india.format()} ;
+      
+    Reminders.create({
+                assetId: fields.assetId,
+                fromUserId: fields.fromUserId,
+                toUserId : fields.toUserId,
+                status: 1,
+                timeZone: fields.timeZone, 
+                Repeat : fields.repeat,
+                Expired : 'No', 
+                deliverAt: deliverAt,
+                lastDeliveredAt: 'Never'
+            }).then(function (Reminders) {
+                if(Reminders) {
+                console.log('new reminder created! ');
+                 res.status(200).send(Reminders);
+                }       
+            }).catch(function (err){
+                console.log('could not create new Reminder!');
+                res.status(500).send('Could not add the reminder, please contact admin ErroCode:' + err.parent.code);   
+            });
+  }))
+});
+
+// Retrieve Reminder by ID
+router.get("/:id", function(req, res) {
+  console.log('In Get reminder for:' + req.params.id);    
+  res.jsonResultOf(Reminders.findById(req.params.id, { deep: true })
+  .then(function(reminder) {
+    if (!reminder) {
+      throw { status: 404 };
+    }
+    // A message may be viewed only by the sender, the receiver, or an admin.
+    if (!req.isAdmin && req.user.id != reminder.fromUserId && req.user.id != reminder.toUserId) {
+      throw { status: 401 };
+    }
+    return reminder;
+  }));
+});
+
+// Delete Reminder by ID
+router.delete("/:id", function(req, res) {
+  res.jsonResultOf(Reminders.findById(req.params.id)
+  .then(function(reminder) {
+    if (!reminder) {
+      throw { status: 404 };
+    }
+    // A message may be deleted only by the sender.
+    if (!req.isAdmin && req.user.id != reminder.fromUserId) {
+      throw { status: 401 };
+    }
+    return Reminders.destroyById(req.params.id);
+  }));
+});
+
+// Update message by ID.
+router.put("/:id", function(req, res) {
+  res.jsonResultOf(new Promise(function(resolve) {
+    var fields = UPDATE_VALIDATOR.prevalidateUpdate(req.body);
+    resolve(Reminders.findById(req.params.id)
+      .then(function(reminder) {
+        if (!reminder) {
+          throw { status: 404 };
+        }
+
+        fields = UPDATE_VALIDATOR.postvalidateUpdate(reminder, fields);
+        console.log(reminder);
+        if (!fields) {
+          return reminder;
+        }
+        
+        var date = new Date(fields.deliverAt);
+        
+        Reminders.updateAttributes({

```

Comment:
Again, drop the redundant then() call here and simply do:

    resolve(Reminders.updateAttributes({...});

Also, once the fields have been validated, there is no need to handpick individual fields to update.  I may have done that in one place, but that was a special case.   The intent is that you can simply do:

   resolve(Reminders.updateAttributes(fields));